### PR TITLE
binary_search for typescript implemented

### LINKED ISSUE #367: BinarySearch in TypeScript
#### This issue is about [Algo/DS Name](link to a resource for the Algo/DS)

-   [x] Issue Title is in form `[Algo/DS Name] in [Language]` (ex:- BubbleSort in Java)

-   [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate

-   [x] Assign this issue to me (I want to work on this)

   -   [x] I ensure that I am not already assigned to 2 or more issues
  
-   [x] I will send my PR only after I'm assigned to this issue by any maintainer

### PR DESCRIPTION
**Pull request by sachinthaR**
**Fixes #367 **

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes. (You can merge if everything fine)

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-bsrh/typescript/binary_search.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+function binarySearch(item_list, item) {
+    let first = 0;
+    let last = item_list.length - 1;
+    let found = false;
+
+    while (first <= last && !found) {
+        let mid = Math.floor((first + last) / 2);
+        if (item_list[mid] == item) {
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/typescript/binary_search.ts:8]: Codacy found an issue: [Identifier 'item_list' is not in camel case.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6274005)

File: algorithms/ar-bsrh/typescript/binary_search.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+function binarySearch(item_list, item) {
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/typescript/binary_search.ts:1]: Codacy found an issue: [Function 'binarySearch' has a complexity of 5. Maximum allowed is 4.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6274005)

File: algorithms/ar-bsrh/typescript/binary_search.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    let first = 0;
+    let last = item_list.length - 1;
+    let found = false;
+
+    while (first <= last && !found) {
+        let mid = Math.floor((first + last) / 2);
+        if (item_list[mid] == item) {
+            found = true;
+        } else {
+            if (item < item_list[mid]) {
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/typescript/binary_search.ts:11]: Codacy found an issue: [Identifier 'item_list' is not in camel case.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6274005)

File: algorithms/ar-bsrh/typescript/binary_search.ts (hunk 2 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/typescript/binary_search.ts:1]: Codacy found an issue: [Identifier 'item_list' is not in camel case.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6274005)

File: algorithms/ar-bsrh/typescript/binary_search.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+function binarySearch(item_list, item) {
+    let first = 0;
+    let last = item_list.length - 1;
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/typescript/binary_search.ts:3]: Codacy found an issue: [Identifier 'item_list' is not in camel case.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6274005)

File: algorithms/ar-bsrh/typescript/binary_search.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+                first = mid + 1;
+            }
+        }
+    }
+
+    return found;
+}
+
+console.log(binarySearch([1, 2, 3, 5, 8], 6));
+console.log(binarySearch([1, 2, 3, 5, 8], 2));
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/typescript/binary_search.ts:23]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6274005)

File: algorithms/ar-bsrh/typescript/binary_search.ts (hunk 1 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/typescript/binary_search.ts:8]: Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6274005)

Comment (RavinduSachintha) [inline algorithms/ar-bsrh/typescript/binary_search.ts:8]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6274005)

File: algorithms/ar-bsrh/typescript/binary_search.ts (hunk 3 above)
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/typescript/binary_search.ts:11]: Codacy found an issue: [Generic Object Injection Sink](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6274005)

File: algorithms/ar-bsrh/typescript/binary_search.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+            } else {
+                first = mid + 1;
+            }
+        }
+    }
+
+    return found;
+}
+
+console.log(binarySearch([1, 2, 3, 5, 8], 6));
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/typescript/binary_search.ts:22]: Codacy found an issue: [Unexpected console statement.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6274005)

File: algorithms/ar-bsrh/typescript/binary_search.ts (hunk 1 above)
Comment (sachinthaR) [reply algorithms/ar-bsrh/typescript/binary_search.ts:8]: So would it be fixed, if I change item_list to itemList ?

File: algorithms/ar-bsrh/typescript/binary_search.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
+function binarySearch(itemList: number[], item: number) {
```
Comment (RavinduSachintha) [inline algorithms/ar-bsrh/typescript/binary_search.ts:1]: Codacy found an issue: [Function 'binarySearch' has a complexity of 5. Maximum allowed is 4.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6274005)

**Comment by RavinduSachintha** [comment]
@all-contributors please add @sachinthaR for code

**Pull request merged**